### ISSUE TITLE
when adding module annotation and build project error shown

### ISSUE THREAD

**Issue by michaelsam94** [google/dagger#2845]
`Execution failed for task ':app:kaptDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.gradle.internal.KaptWithoutKotlincTask$KaptExecutionWorkAction
> java.lang.reflect.InvocationTargetException (no error message)

* Try:
Run with --stacktrace option to get the stack trace.  Run with --scan to get full insights.`

**Comment by michaelsam94** [comment]
I create repo showing the error [https://github.com/michaelsam94/dagger_example](url)

**Comment by danysantiago** [comment]
That error is usually an indicator that the annotation processor crashed, to see the full error use the `--stacktrace` option, like this `./gradlew assembleDebug`.

Then you'll see the full error:
```
Caused by: java.lang.IllegalStateException: Unsupported metadata version. Check that your Kotlin version is >= 1.0
        at dagger.internal.codegen.kotlin.KotlinMetadata.metadataOf(KotlinMetadata.java:216)
        at dagger.internal.codegen.kotlin.KotlinMetadata.from(KotlinMetadata.java:194)
        at dagger.internal.codegen.kotlin.KotlinMetadataFactory.create(KotlinMetadataFactory.java:56)
        at dagger.internal.codegen.kotlin.KotlinMetadataUtil.isCompanionObjectClass(KotlinMetadataUtil.java:86)
        at dagger.internal.codegen.ModuleProcessingStep.process(ModuleProcessingStep.java:116)
        at dagger.internal.codegen.ModuleProcessingStep.process(ModuleProcessingStep.java:59)
        at dagger.internal.codegen.validation.TypeCheckingProcessingStep.lambda$process$0(TypeCheckingProcessingStep.java:51)
        at com.google.common.collect.SingletonImmutableBiMap.forEach(SingletonImmutableBiMap.java:65)
        at dagger.internal.codegen.validation.TypeCheckingProcessingStep.process(TypeCheckingProcessingStep.java:48)
        at dagger.internal.codegen.ModuleProcessingStep.process(ModuleProcessingStep.java:104)
        at dagger.internal.codegen.ModuleProcessingStep.process(ModuleProcessingStep.java:59)
        at dagger.internal.codegen.statistics.DaggerStatisticsCollectingProcessingStep.process(DaggerStatisticsCollectingProcessingStep.java:52)
        at dagger.shaded.auto.common.BasicAnnotationProcessor.process(BasicAnnotationProcessor.java:330)
        at dagger.shaded.auto.common.BasicAnnotationProcessor.process(BasicAnnotationProcessor.java:181)
        at org.jetbrains.kotlin.kapt3.base.incremental.IncrementalProcessor.process(incrementalProcessors.kt:90)
        at org.jetbrains.kotlin.kapt3.base.ProcessorWrapper.process(annotationProcessing.kt:175)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:980)
        ... 41 more
```

Anyway, this is an issue with Kotlin 1.5.0 and it was fixed in [Dagger 2.34](https://github.com/google/dagger/releases/tag/dagger-2.34), specifically you are hitting https://github.com/google/dagger/issues/2511. I see the sample project is using 2.27, please update your Dagger version.
